app.rb

```diff
@@ -0,0 +1,137 @@
+require './book'
+require './person'
+require './student'
+require './teacher'
+
+# create app class
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name:#{person.name} ID:#{person.id} Age:#{person.age}"
+    end
+  end
+
+  def create_person
+    puts "\nCreate a person (1) Student or (2) Teacher?"
+    option = gets.chomp.to_i
+
+    print 'Age:'
+    age = gets.chomp.to_i
+    print 'Name:'
+    name = gets.chomp
+
+    case option
+    when 1
+      create_student(age, name)
+    when 2
+      create_teacher(age, name)
+    else
+      puts 'Invalid option.'
+    end
+  end
+
```

Comment:
- [ ] @BilalLiaquat7  Great work adding a method to add a new person, however, am getting the error below when I try to create either a student or a teacher, this made it impossible to check if `list all people, `Create a rental`, and `List all rentals for a given person id` are working correctly.
![image](https://github.com/BilalLiaquat7/school_library/assets/61408860/7d61eb67-f58c-40c4-bbcc-d81f41bea480)

Reply:
**[REQUIRED CHANGES]**

- As pointed out by the previous reviewer, a user still gets the same error when trying to create a student or a teacher. Please, kindly fix it so that every other option will be tested and they should work properly.



![image](https://github.com/BilalLiaquat7/school_library/assets/59397159/2b64f3f9-a48b-4fbb-8a2c-1226aa4c06e4)